pkg/ca/fileca/load.go

```diff
@@ -0,0 +1,61 @@
+// Copyright 2021 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package fileca
+
+import (
+	"crypto"
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+)
+
+func loadKeyPair(certPath, keyPath string) (*x509.Certificate, crypto.Signer, error) {
+	tlsCert, err := tls.LoadX509KeyPair(certPath, keyPath)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var (
+		key  crypto.Signer
+		cert *x509.Certificate
+	)
+	{
+		var ok bool
+		key, ok = tlsCert.PrivateKey.(crypto.Signer)
+		if !ok {
+			// Shouldn't be reachable because of the data validation that
+			// tls.LoadX509KeyPair already did
+			return nil, nil, errors.New(`fileca: loaded private key can't be used to sign`)
+		}
+
+		if len(tlsCert.Certificate) != 1 {
+			return nil, nil, errors.New(`fileca: expected certificate chain with exactly one cert`)
```

Comment:
I think this needs to be changed to support a chain of certificates instead of a single certificate.

In the example with an intermediate certificate being issued off a root, how else do we include the root to pass back to cosign?

Reply:
That's a good catch.  Perhaps we can add this and a test in a follow-up?

Reply:
Ahhh ok thanks! This is just my weak x509 knowledge. I'll iterate and add a test case with a certificate chain to load :D Can ya'll point out where the root is passed back to cosign in the code base? For example in the googleca backend?

Reply:
`CodeSigningCertificate` contains the chain. For example, with the CA Service backend, it's generated here - https://github.com/sigstore/fulcio/blob/d77d444350a193906799cad013d98b64f65a3288/pkg/ca/googleca/v1/googleca.go#L161
I'd recommend we do some basic validation of the chain too on initialization, to make sure that the intermediates all chain up to the root.